optuna/integration/mxnet.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import absolute_import
+
+import mxnet as mx  # NOQA
+import optuna
+
+
+class MxnetPruningCallback(object):
+    """Mxnet callback to prune unpromising trials.
+
+    Example:
+
+        Add a pruning callback which observes validation losses.
+
+        .. code::
+
+            model.fit(X, y, batch_end_callback=MxnetPruningCallback(trial, eval_metric='acc'))
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
+        monitor:
+            An evaluation metric for pruning, e.g., ``cross-entropy`` and
+            ``accuracy``. Please refer to `keras.Callback reference
+            <https://keras.io/callbacks/#callback>`_ for further details.
+    """
+
+    def __init__(self, trial, eval_metric='accuracy'):
+        # type: (optuna.trial.Trial, str) -> None
+
+        self.trial = trial
+        self.eval_metric = eval_metric
+
+    def __call__(self, param):
+        # type: (mx.model.BatchEndParams,) -> None
+
+        if param.nbatch == 1 and param.eval_metric is not None:
```

Comment:
Could you explain why the condition `param.nbatch == 1` is needed? 
(I'm unfamiliar with MXNet ...)

Reply:
The callback is triggered for every batch in an epoch. If we want the pruning check to happen every epoch, this condition becomes necessary.

Reply:
I see, thanks.
However, I'm worried about the current implementation seems to report partial evaluation results (i.e., the result of the first batch of each epoch). This might adversely affect the decisions of pruning.
I think other callbacks (e.g., `eval_end_callback`) can be used for pruning purpose (I'm not sure though).

Reply:
I have modified the callback to be used with eval_end_callback. Also updated the test for it.

Reply:
Thanks.